lib/oli_web/live/delivery/student/learn_live.ex

```diff
@@ -20,7 +21,12 @@ defmodule OliWeb.Delivery.Student.LearnLive do
           socket.assigns[:current_user]
         )
 
-    root_node = full_hierarchy(socket.assigns.section)
+    section = socket.assigns.section
+
+    root_node =
+      SectionCache.get_or_compute(section.slug, :full_hierarchy, fn ->
+        full_hierarchy(section)
+      end)
```

Comment:
Does it make sense to have a `full_hierarchy/1` function in the Sections context? That function could wrap the get_or_compute logic so anyone could call that function without worrying about whether it was computed or retrieved from the cache.